dis_snek/client.py

```diff
@@ -462,6 +464,46 @@ def wait_for(self, event: str, checks: Optional[Callable[..., bool]] = MISSING,
 
         return asyncio.wait_for(future, timeout)
 
+    async def wait_for_component(
+        self,
+        messages: Union[Message, int, list] = None,
+        components: Union[str, dict, list] = None,
+        check=None,
+        timeout=None,
+    ) -> Component:
+        """
+        Waits for a message to be sent to the bot.
+
+        Args:
+            messages: The message object to check for.
+            components: The components to wait for.
+            check: A predicate to check what to wait for.
+            timeout: The number of seconds to wait before timing out.
+
+        Returns:
+            `Component` that was invoked, or `None` if timed out. Use `.context` to get the `ComponentContext`.
+        """
+        if not (messages or components):
+            raise ValueError("You must specify messages or components (or both)")
+
+        message_ids = list(get_messages_ids(messages)) if messages else None
+        custom_ids = list(get_components_ids(components)) if components else None
+
+        # automatically convert improper custom_ids
+        if custom_ids and not all(isinstance(x, str) for x in custom_ids):
+            custom_ids = [str(i) for i in custom_ids]
+
+        def _check(event: Component):
+            ctx: ComponentContext = event.context
+            if check and not check(event):
```

Comment:
I think custom check callback should ran after checking that message and custom_id matches.

Reply:
did this instead, think thats what you wanted:
```py
def _check(event: Component):
    ctx: ComponentContext = event.context
    # if custom_ids is empty or there is a match
    wanted_message = not message_ids or ctx.origin_message_id in message_ids
    wanted_component = not custom_ids or ctx.custom_id in custom_ids
    if wanted_message and wanted_component:
        if check is None or check(event):
            return True
        return False
    return False
```

Reply:
this worked:
```py
def _check(event: Component):
    ctx: ComponentContext = event.context
    # if custom_ids is empty or there is a match
    wanted_message = not message_ids or ctx.message.id in (
        [message_ids] if isinstance(message_ids, int) else message_ids
    )
    wanted_component = not custom_ids or ctx.custom_id in custom_ids
    if wanted_message and wanted_component:
        if check is None or check(event):
            return True
        return False
    return False
```